combat-trainer.lic

```diff
@@ -355,11 +355,25 @@ class LootProcess
   end
 
   def should_perform_ritual(prepare_consume)
-    !prepare_consume &&
-      ((@necro_train_first_aid && DRSkill.getxp('First Aid') < 32) || DRSkill.getxp('Thanatology') < 32)
+   !prepare_consume &&
+    ((@ritual_type == "dissect" && DRSkill.getxp('First Aid') < 32) ||
+    (@ritual_type == "harvest" && DRSkill.getxp('Skinning') < 32) ||
+    DRSkill.getxp('Thanatology') < 32)
+  end
+
+  def check_ritual_xp
+    DRSkill.getxp('First Aid') > 31 && DRSkill.getxp('Skinning') > 31
+  end
+
+  def cycle_rituals
```

Comment:
I can clean try to clean it up a bit but I don't think its a good idea to combine them.  The consume logic is already pretty messy.  If I layered the cycle on top of that it would be even harder to manage.

Plus, not everyone will want to use the cycle ritual (unless maybe you are suggesting additional default behavior) so I don't really see that we would need to force the cycle calculations by combining everything into one piece.